### DISCUSSION TITLE
Awesome structure for NextJs 13.4.4 server actions

### DISCUSSION THREAD

**Discussion by gukii** [airjp73/rvf#299] [General]
This is a really delightful library. I wish it was available during my last Remix project.

Now, working with the new server actions and functions of NextJs 13.4.4. I find that so much could be covered by the structure and basis you provide here..

Having tried NextJs' server actions with server rendered forms, validation, error messages, etc. I got most of these things working in basic NextJs server actions code. The fact that remix is built on so many common standards and browser paradigms, it really feels like technology is converging..

Sorry for the NextJs-centric post, just wanted to get that out there. Please remove the post if it doesn t fit in here.

**Comment by pigoz** [comment]
It looks like the library is not using many Remix specifics. Could probably be easily ported.

**Comment by shawnmclean** [comment]
Has anyone figured out how to use with useActionState or useFormState from react?

**Reply by shawnmclean** [reply]
One thing I've found, was the use of zod.

The reason why I found this library was that I was looking for something to handle and parse `FormData`.

`withZod` works for me in the server actions for validating and parsing out this data.

Nice work on the library @airjp73.

In your docs, its not only useful for remix server side, an example can be added for nextjs server actions receiving `FormData` and throw some keywords on it to bring in folks who are looking for this but don't want a heavy library like react-hook-form.
